phpBB/includes/ucp/ucp_main.php

```diff
@@ -299,6 +313,22 @@ function main($id, $mode)
 						$tracking_topics = ($tracking_topics) ? tracking_unserialize($tracking_topics) : array();
 					}
 
+					/**
+					 * Modify the query used to retrieve a list of subscribed forums
+					 *
+					 * @event core.ucp_main_subscribed_forums_modify_query
+					 * @var array	sql_array	       The subscribed forums query
+					 * @var array   forbidden_forums   The list of forbidden forums
+					 * @var user    user               The current user
```

Comment:
Classes like the user class shouldn't be passed with an event. They will be copied which causes some overhead that is really not needed at all. Instead, these classes should be passed to the events via the container definition.